pkg/k8sutil/create.go

```diff
@@ -161,3 +166,67 @@ func parseJSONManifest(data []byte) ([]manifest, error) {
 	}
 	return manifests, nil
 }
+
+// UpdateNamespace updates the namespace.
+func UpdateNamespace(namespace Namespace, kubeconfig []byte) error {
+	cs, err := NewClientset(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("creating clientset: %w", err)
+	}
+
+	if namespace.Name == "" {
+		return fmt.Errorf("namespace name can't be empty")
+	}
+
+	ns, err := cs.CoreV1().Namespaces().Get(context.TODO(), namespace.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	existingLabels := ns.ObjectMeta.Labels
+	existingAnnotations := ns.ObjectMeta.Annotations
+	// Delete keys from the existing labels and annotations that are defined by Lokomotive
+	// ie. keys with the prefix `lokomotive.kinvolk.io`.
+	// This ensures that when the namespace is updated, labels added by the user
+	// are retained in the update.
+	for key := range existingLabels {
+		if strings.Contains(key, "lokomotive.kinvolk.io") {
+			delete(existingLabels, key)
+		}
+	}
+
+	for key := range existingAnnotations {
+		if strings.Contains(key, "lokomotive.kinvolk.io") {
+			delete(existingAnnotations, key)
+		}
+	}
+
+	labels := namespace.Labels
+	for key, value := range existingLabels {
+		labels[key] = value
+	}
+
+	annotations := namespace.Annotations
+	// initialize map if nil
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	for key, value := range existingAnnotations {
+		annotations[key] = value
+	}
+
+	// Update the namespace.
+	_, err = cs.CoreV1().Namespaces().Update(context.TODO(), &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        namespace.Name,
+			Labels:      labels,
+			Annotations: annotations,
+		},
+	}, metav1.UpdateOptions{})
```

Comment:
I'd simplify this code like this:
```suggestion
	// Update the namespace.
	_, err = cs.CoreV1().Namespaces().Update(context.TODO(), &v1.Namespace{
		ObjectMeta: metav1.ObjectMeta{
			Name:        namespace.Name,
			Labels:      somerandompackage.MergeMaps(namespace.Labels, ns.ObjectMeta.Labels),
			Annotations: somerandompackage.MergeMaps(namespace.Annotations, ns.ObjectMeta.Annotations),
		},
	}, metav1.UpdateOptions{})
```

Reply:
`somerandompackage` ?  perhaps `internal` package ?

Reply:
I think its fine as it is.

Reply:
added.